sources/packages/web/src/components/aest/students/modals/EditStudentProfileModal.vue

```diff
@@ -0,0 +1,150 @@
+<template>
+  <v-form ref="editStudentProfileForm">
+    <modal-dialog-base
+      title="Edit Student Profile"
+      sub-title="Enter the information to be updated below."
+      :showDialog="showDialog"
+    >
+      <template #content>
+        <title-value
+          ><template #value
+            ><v-text-field
+              hide-details="auto"
+              label="Given Names"
+              v-model="formModel.givenNames"
+              variant="outlined"
+              :rules="[checkGivenNameLengthRule]"
+              class="mb-6" />
+            <v-spacer /></template
+        ></title-value>
+        <title-value
+          ><template #value
+            ><v-text-field
+              hide-details="auto"
+              label="Lastname"
+              v-model="formModel.lastName"
+              variant="outlined"
+              :rules="[checkLastNameLengthRule]"
+              class="mb-6" />
+            <v-spacer /></template
+        ></title-value>
+        <title-value
+          ><template #value
+            ><v-text-field
+              density="compact"
+              label="Date of Birth"
+              variant="outlined"
+              v-model="formModel.birthdate"
+              type="date"
+              :max="getTodaysDate()"
+              hide-details="auto"
+              class="mb-6" />
+            <v-spacer /></template
+        ></title-value>
+        <title-value
+          ><template #value
+            ><v-text-field
+              hide-details="auto"
+              label="Email"
+              v-model="formModel.email"
+              variant="outlined"
+              :rules="[checkEmailLengthRule, checkEmailValidationRule]"
+              class="mb-6" />
+            <v-spacer /></template
+        ></title-value>
+      </template>
+      <template #footer>
+        <check-permission-role :role="Role.StudentEditProfile">
+          <template #="{ notAllowed }">
```

Comment:
No need to block the button at this stage since the access to the entire modal is already blocked.